mainapp/views.py

```diff
@@ -168,11 +170,16 @@ class Maintenance(TemplateView):
 
 
 def mapdata(request):
-    if("district" in request.GET.keys()):
-        data = Request.objects.exclude(latlng__exact="").filter(district = request.GET.get("district")).values() 
+    district = request.GET.get("district", "all")
+    data = cache.get("mapdata:" + district)
+    if data:
+        return JsonResponse(list(data) , safe=False)
+    if district != "all":
+        data = Request.objects.exclude(latlng__exact="").filter(district=district).values()
+        cache.set("mapdata:" + district, data, settings.CACHE_TIMEOUT)
     else:
         data = Request.objects.exclude(latlng__exact="").values()
-
+        cache.set("mapdata:" + district, data, settings.CACHE_TIMEOUT)
```

Comment:
You can take this out side of `if else` block since you have same code for both cases

Reply:
ok